route_info/src/lib.rs

```diff
@@ -88,8 +82,82 @@ impl RouteInfo {
             .collect::<BTreeSet<_>>();
         abstutil::plain_list_names(road_names)
     }
+
+    /// Given the JSON waypoints array produced by route-snapper, generate GeoJSON LineStrings
+    /// covering each segment of the route where `speed_limit` (in mph) is defined. Freehand and
+    /// unknown segments are not returned.
+    #[wasm_bindgen(js_name = speedLimitForRoute)]
+    pub fn speed_limit_for_route(&self, raw_waypoints: JsValue) -> Result<String, JsValue> {
+        let raw_waypoints: Vec<RawRouteWaypoint> = serde_wasm_bindgen::from_value(raw_waypoints)?;
+        let waypoints = self.parse_waypoints(raw_waypoints).map_err(err_to_js)?;
+
+        let mut features = Vec::new();
+        for pair in waypoints.windows(2) {
+            if let (Waypoint::Snapped(_, i1), Waypoint::Snapped(_, i2)) = (&pair[0], &pair[1]) {
+                if let Some(path) = self.geometric_path(*i1, *i2) {
+                    // Add one LineString per Road where speed_limit is defined.
+                    // We could try to glue together contiguous LineStrings where the speed is the
```

Comment:
A previous commit in this PR did things in a much more complicated way. It's preserved in git if we ever need something like it again.